### PR TITLE
[GCS] No need to send actor death signal from gcs client

### PR DESCRIPTION
**Pull request by micafan**
GCS Client will send actor died signal to notify python worker that an actor has dead. It seems that python worker does not need this signal now, so GCS Client can remove these signal logic.

## Related issue number

#7675

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by mehrdadn** [comment]
Thanks!! Before we merge this, I'll try to see if I can get it working on Windows. It doesn't seem to run as-is yet. (It fails to connect to Redis at all.)

**Comment by mehrdadn** [comment]
Never mind, it's working fine, I was making a mistake somewhere when testing :) thanks!

**Review by raulchen — APPROVED** [review]

**Pull request merged**
